server/config/config.go

```diff
@@ -79,3 +90,49 @@ func (c Config) SetContextDefaultsForApp(appID apps.AppID, cc *apps.Context) *ap
 func (c Config) AppPath(appID apps.AppID) string {
 	return c.PluginURL + PathApps + "/" + string(appID)
 }
+
+func (c *Config) Reconfigure(stored StoredConfig, mmconf *model.Config) error {
+	mattermostSiteURL := mmconf.ServiceSettings.SiteURL
+	if mattermostSiteURL == nil {
+		return errors.New("plugin requires Mattermost Site URL to be set")
+	}
+	mattermostURL, err := url.Parse(*mattermostSiteURL)
+	if err != nil {
+		return err
+	}
+
+	c.StoredConfig = stored
+
+	c.MattermostSiteURL = *mattermostSiteURL
+	c.MattermostSiteHostname = mattermostURL.Hostname()
+	c.PluginURLPath = "/plugins/" + c.BuildConfig.Manifest.Id
+	c.PluginURL = strings.TrimRight(*mattermostSiteURL, "/") + c.PluginURLPath
+
+	c.MaxWebhookSize = 75 * 1024 * 1024 // 75Mb
+	if mmconf.FileSettings.MaxFileSize != nil {
+		c.MaxWebhookSize = *mmconf.FileSettings.MaxFileSize
+	}
+
+	c.DeveloperMode = pluginapi.IsConfiguredForDevelopment(mmconf)
+
+	// use CloudAccessEnvVar for now to detect the Cloud mode
+	cloudAccessKey := os.Getenv(awsapps.CloudAccessEnvVar)
+	c.MattermostCloudMode = false
+	if cloudAccessKey != "" {
+		c.MattermostCloudMode = true
+	}
```

Comment:
https://github.com/mattermost/mattermost-server/blob/c90c371fd4016345e4fcc2c619d3b5c47edaeecc/model/license.go#L86 should be used to decide if cloud is used

Reply:
Done